cmd/krew/cmd/list.go

```diff
@@ -56,17 +136,44 @@ Remarks:
 				return nil
 			}
 
-			// print table
-			var rows [][]string
-			for p, version := range plugins {
-				rows = append(rows, []string{p, version})
+			if *output.OutputFormat == "wide" || (*output.OutputFormat == "" && isTerminal(os.Stdout)) {
+				// print table
+				var rows [][]string
+				for p, version := range plugins {
+					rows = append(rows, []string{p, version})
+				}
+				rows = sortByFirstColumn(rows)
+				return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
 			}
-			rows = sortByFirstColumn(rows)
-			return printTable(os.Stdout, []string{"PLUGIN", "VERSION"}, rows)
+
+			if slice.ContainsString(output.AllowedFormats(), *output.OutputFormat, nil) {
+				objs := make([]Entry, 0, len(plugins))
+				for plugin, version := range plugins {
+					obj := Entry{plugin, version}
+					objs = append(objs, obj)
+				}
+				objs = sortByName(objs)
+				for _, obj := range objs {
```

Comment:
The proper way to print lists with the object printer is to pass it the full list and let the object printer handle lists. This is important when you want to read that output in a structured way.

For example, when printing the list as json, the code will produce several json objects like so:
```json
{ "name": "krew", "version": "v0.3.1" }
{ "name": "cssh", "version": "v1.1.1" }
```
instead of
```json
{
    "apiVersion": "v1",
    "kind": "List",
    "items": [
      { "name": "krew", "version": "v0.3.1" },
      { "name": "cssh", "version": "v1.1.1" }
    ]
}
```
The latter should be the preferred format for machine processing.

For that to work properly, you can do something like this https://github.com/corneliusweig/ketall/blob/07fd92429ca33385012971f9cc2a1408c1db6cdd/pkg/ketall/printer/adapter.go#L56

Reply:
I made a new type to encapsulate []Entry and implement runtime.Object so an array of objects is printed.